bin/postgres-ha/bootstrap/pre-bootstrap.sh

```diff
@@ -318,8 +318,46 @@ build_bootstrap_config_file() {
     # postgresql.conf file
     if [[ "${PGHA_TLS_ENABLED}" == "true" ]]
     then
+      echo_info "Enabling TLS"
+      pghba_tls_file="/opt/cpm/conf/postgres-ha-pghba-tls.yaml"
+
+      # if there is a TLS keypair for a replication user detected, we will need
+      # to copy this to a special mount in order to properly enable
+      # certificate-based authentication between replicas. This is due to
+      # PostgreSQL requiring a client's TLS key to have permissions no less strict
+      # than 0600. In this case, the client is the "postgres" user, and because
+      # Kubernetes Secrets are owned by the group, we need to create a copy of
+      # this key with the correct permissions
+      if [[ -f "/pgconf/tls/tls-replication.key" ]] && [[ -f "/pgconf/tls/tls-replication.crt" ]]
+      then
+        echo_info "Enabling certificate-based authentication for replication"
+        touch /pgconf/tls-replication/tls.key /pgconf/tls-replication/tls.crt
+        chown postgres:postgres /pgconf/tls-replication/tls.key /pgconf/tls-replication/tls.crt
+        chmod 0600 /pgconf/tls-replication/tls.key /pgconf/tls-replication/tls.crt
+        cat /pgconf/tls/tls-replication.key > /pgconf/tls-replication/tls.key
+        cat /pgconf/tls/tls-replication.crt > /pgconf/tls-replication/tls.crt
```

Comment:
I expected the origin to be shaped like a Kubernetes TLS Secret (one directory containing `tls.key` and `tls.crt`).

Is the destination shaped this way in case the Secret can be mounted directly in the future?

Reply:
it is. it's included as part of a projected volume that has another TLS secret. You need to view the corresponding patch at CrunchyData/postgres-operator#1662

Reply:
Leaving this as is.